tests/v2/integration/catalogv2/ui_plugin_test.go

```diff
@@ -0,0 +1,369 @@
+	rv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/dashboard/plugin"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/clients/rancher/catalog"
+	client "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	stevev1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/repo"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"net/http"
+	"testing"
+	"time"
+)
+
+type UIPluginTest struct {
+	suite.Suite
+	client           *rancher.Client
+	session          *session.Session
+	restClientGetter genericclioptions.RESTClientGetter
+	catalogClient    *catalog.Client
+	cluster          *client.Cluster
+	corev1           corev1.CoreV1Interface
+	originalBranch   string
+	originalGitRepo  string
+}
+
+func (w *UIPluginTest) TearDownSuite() {
+	w.Require().NoError(w.uninstallApp(namespace.UIPluginNamespace, "uk-locale"))
+	w.Require().NoError(w.uninstallApp(namespace.UIPluginNamespace, "clock"))
+	w.Require().NoError(w.catalogClient.ClusterRepos().Delete(context.Background(), "extensions-examples", metav1.DeleteOptions{PropagationPolicy: &propagation}))
+	w.session.Cleanup()
+
+}
+
+func (w *UIPluginTest) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+	insecure := true
+	w.client.RancherConfig.Insecure = &insecure
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(w.client, "local")
+	require.NoError(w.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(w.T(), err)
+
+	//restConfig.Insecure = true
+
+	cset, err := kubernetes.NewForConfig(restConfig)
+	require.NoError(w.T(), err)
+	w.corev1 = cset.CoreV1()
+
+	w.restClientGetter, err = kubeconfig.NewRestGetter(restConfig, *kubeConfig)
+	require.NoError(w.T(), err)
+	require.NoError(w.T(), err)
+	_, err = w.catalogClient.ClusterRepos().Create(context.Background(), &rv1.ClusterRepo{
+		ObjectMeta: metav1.ObjectMeta{Name: "extensions-examples"},
+		Spec:       rv1.RepoSpec{GitRepo: "https://github.com/diogoasouza/ui-plugin-examples", GitBranch: "main"},
```

Comment:
We can't use private repos.

Reply:
Waiting for https://github.com/rancher/ui-plugin-examples/pull/37 to be merged